src/components/CreateVmWizard/steps/BasicSettings.js

```diff
@@ -289,6 +305,25 @@ class BasicSettings extends React.Component {
         }
         break
 
+      case 'enableSysPrepTimezone': // Configure Timezone checkbox change
+        changes[field] = value
+        // if the Configure Timezone checkbox checked, sysprep timezone should be set, otherwise reset it to the default/no value
+        changes.initTimezone = value ? this.props.data.initTimezone || this.props.data.selectedSysprepTimezone : ''
```

Comment:
We do need it like this, it is because you need to remember it, when moving between the steps of the wizard, especially from other steps back to the Basic settings step. Without pushing it to `props.data` the information about the last selected sysprep timezone will be lost.

Reply:
It seems that `setState()` should work exactly as you describe.  I assume that the step object is not destroyed until the wizard is done.  `props.data` as pointed by @bamsalem already have a structure defined by `BASIC_DATA_SHAPE`.

Reply:
> I assume that the step object is not destroyed until the wizard is done. props.data as pointed by @bamsalem already have a structure defined by BASIC_DATA_SHAPE.

The step object is destroyed  so we cannot store there any data. All the data need to be stored in the parent component.
Taking this into consideration I would change BASIC_DATA_SHAPE and add both props stored there (`lastInitTimezone` and `enableInitTimezone`) as required props.
They could be optional as well but then we won't get much type safety.

Reply:
Other props are also not defined properly in the parent or via PropType.
Let leave it for another PR.